jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/EppsteinShortestPathIterator.java

```diff
@@ -0,0 +1,747 @@
+/*
+ * (C) Copyright 2019-2019, by Semen Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.shortestpath;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.GraphType;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.util.Pair;
+import org.jgrapht.graph.EdgeReversedGraph;
+import org.jgrapht.traverse.DepthFirstIterator;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+
+/**
+ * Iterator over the shortest paths (not required to be simple) between two vertices in a graph sorted by weight.
+ *
+ * <p>
+ * This implementation can only be used for directed simple graphs.
+ * Also for this iterator to work correctly the graph must not be modified during iteration.
+ * Currently there are no means to ensure that, nor to fail-fast. The results of such
+ * modifications are undefined.
+ *
+ * <p>
+ * First the shortest paths tree in the edge reversed graph starting at
+ * {@code sink} is built. Thus we get distances $d(v)$ from every vertex
+ * $v$ to {@code sink}. We then define a sidetrack edge to be an edge, which
+ * is not in the shortest paths tree but which source and target belong to it.
```

Comment:
do you mean "but which includes either the source or target"?

Reply:
Here I mean that a sidetrack edge is an edge which does not belong to the shortest path tree described before, but the vertices it starts and ends at belong to "branches" of the tree. The source and target of the sidetrack collide with the source and the target of paths we a looking for in the algorithm. So I will remove this part of sentence because if is not needed.